src/reporter/github.js

```diff
@@ -69,40 +115,42 @@ export default class GitHub {
   }
 
   async createIssue({ title, description: body, labels }) {
+    await this.ensureCacheLoaded();
+
     const { data: issue } = await this.octokit.request('POST /repos/{owner}/{repo}/issues', {
       ...this.commonParams,
       title,
       body,
       labels,
     });
 
+    this.issuesCache.set(issue.title, issue);
+
     return issue;
   }
 
   async updateIssue(issue, { state, labels }) {
+    await this.ensureCacheLoaded();
+
     const { data: updatedIssue } = await this.octokit.request('PATCH /repos/{owner}/{repo}/issues/{issue_number}', {
       ...this.commonParams,
       issue_number: issue.number,
       state,
       labels,
     });
 
+    this.issuesCache.set(updatedIssue.title, updatedIssue);
+
     return updatedIssue;
   }
 
-  async getIssue({ title, ...searchParams }) {
-    const issues = await this.octokit.paginate('GET /repos/{owner}/{repo}/issues', {
-      ...this.commonParams,
-      per_page: 100,
-      ...searchParams,
-    }, response => response.data);
-
-    const [issue] = issues.filter(item => item.title === title); // Since only one is expected, use the first one
-
-    return issue;
+  getIssue(title) {
+    return this.issuesCache.get(title) || null;
```

Comment:
Why isn't `undefined` acceptable as a return value if the issue is not found in the cache? 🙂 
```suggestion
    return this.issuesCache.get(title);
```